src/main/java/io/cdap/plugin/gcp/common/GCPConfig.java

```diff
@@ -63,7 +63,8 @@ public String getProject() {
   @Nullable
   public String tryGetProject() {
     if (containsMacro(NAME_PROJECT) && Strings.isNullOrEmpty(project)) {
-      return null;
+      // macro substitution has not been performed yet, so return raw value for validation
+      return getRawProperties().getProperties().get(NAME_PROJECT);
```

Comment:
this is not the right fix. Whatever is trying to do validation with the project should make sure it's not a macro before doing the check.

Reply:
+1 to what Albert mentioned